### PR TITLE
Fix Gnu compilation on Hera

### LINKED ISSUE #962: GNU build on Hera is failing
The head of develop (2794d41) no longer compiles on Hera with Gnu. I get this error:

```
Lmod has detected the following error:  The following module(s) are unknown: "openmpi/4.1.5"

Please check the spelling or version number. Also try "module spider ..."
It is also possible your cache file is out-of-date; it may help to try:
  $ module --ignore_cache load "openmpi/4.1.5"

Also make sure that all modulefiles written in TCL start with the string #%Module

Executing this command requires loading "openmpi/4.1.5" which failed while processing the following module(s):

    Module fullname      Module Filename
    ---------------      ---------------
    stack-openmpi/4.1.5  /scratch1/NCEPDEV/nems/role.epic/spack-stack/spack-stack-1.6.0/envs/unified-env-rocky8/install/modulefiles/gcc/9.2.0/stack-openmpi/4.1.5.lua
    build.hera.gnu       /scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/UFS_UTILS.upstream/modulefiles/build.hera.gnu.lua
While processing the following module(s):
    Module fullname      Module Filename
    ---------------      ---------------
    stack-openmpi/4.1.5  /scratch1/NCEPDEV/nems/role.epic/spack-stack/spack-stack-1.6.0/envs/unified-env-rocky8/install/modulefiles/gcc/9.2.0/stack-openmpi/4.1.5.lua
    build.hera.gnu       /scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/UFS_UTILS.upstream/modulefiles/build.hera.gnu.lua
```

### PR DESCRIPTION
**Pull request by GeorgeGayno-NOAA**
## DESCRIPTION OF CHANGES: 
- Update the Gnu build module on Hera to point to an updated stack.
- Fix some namelist read logic in `global_cycle` and `cpld_gridgen` so it works with Gnu.
- Minor fix to a preprocessor directive in `chgres_cube` that Gnu flagged as a warning.

## TESTS CONDUCTED: 

- [x] Compile branch on all Tier 1 machines using Intel. Done using 05b6fc2.
- [x] Compile branch on Hera using GNU. Done using 4dca77a.
- [x] Compile branch in 'Debug' mode on WCOSS2. Done on Dogwood using 05b6fc2.
- [x] Run unit tests locally on any Tier 1 machine. Done on Hera using Gnu (4dca77a).
- [x] Run `global_cycle` consistency tests locally on all Tier 1 machines (using Intel). Done using 05b6fc2. All tests passed as expected.

All consistency tests were run using Gnu on Hera using 4dca77a. All tests except one `global_cycle` test ran to completion. The baselines did not always match, but that is expected since they were created using Intel. For more details, see: https://github.com/ufs-community/UFS_UTILS/issues/962#issuecomment-2206158835

The `cpld_gridgen` consistency test was run using Intel on Hercules (05b6fc2). All tests passed as expected.

## DEPENDENCIES:
None.

## DOCUMENTATION:
N/A

## ISSUE: 
Fixes #962.

### CODE REVIEW

File: modulefiles/build.hera.gnu.lua [hunk 1]
```diff
@@ -41,7 +41,7 @@ load(pathJoin("w3emc", w3emc_ver))
 sigio_ver=os.getenv("sigio_ver") or "2.3.2"
 load(pathJoin("sigio", sigio_ver))
 
-hdf5_ver=os.getenv("hdf5_ver") or "1.14.0"
+hdf5_ver=os.getenv("hdf5_ver") or "1.14.3_gnu9.2.0"
```
Comment (AlexanderRichert-NOAA) [inline modulefiles/build.hera.gnu.lua]: Where does this '_gnu9.2.0' come from? Is this one of the system-provided modules?

Comment (GeorgeGayno-NOAA) [reply modulefiles/build.hera.gnu.lua]: That module load does not appear to do anything. When I remove it (4dca77a), hdf5 v1.14.0 is still loaded.  

I can also remove the netcdf loads from the build module with no effect. Both libraries must be loaded via other modules. 

I will rerun my Gnu tests using 4dca77a.

Comment (GeorgeGayno-NOAA) [reply modulefiles/build.hera.gnu.lua]: Reran all my Gnu tests. Results did not change: https://github.com/ufs-community/UFS_UTILS/issues/962#issuecomment-2206158835

Comment (GeorgeGayno-NOAA) [reply modulefiles/build.hera.gnu.lua]: @AlexanderRichert-NOAA - any further comments?

**Comment by GeorgeGayno-NOAA** [comment]
@AlexanderRichert-NOAA - I would like to merge. Any other comments?

**Review by AlexanderRichert-NOAA — APPROVED** [review]
@GeorgeGayno-NOAA no other comments

**Pull request merged**